lib/route.js

```diff
@@ -289,6 +289,7 @@ function buildRouting (options) {
     req.originalUrl = req.url
     var hostname = req.headers['host']
     var ip = req.connection.remoteAddress
+    var proto = req.connection.encrypted ? 'https' : 'http'
```

Comment:
The protocol should also contain `:`, so we mimic the behavior of [`URL`](https://nodejs.org/api/url.html#url_the_whatwg_url_api), what do you think?

```sh
> new URL('https://example.com')
URL {
  href: 'https://example.com/',
  origin: 'https://example.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'example.com',
  hostname: 'example.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '' }

```